### PR TITLE
refactor(commercetools)!: refactor token handling in commercetools

### PR DESCRIPTION
**Pull request by filipsobol**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.

#### Changelog
- [X] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [X] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [X] I have written test cases for my code
- [X] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

#### Code standards
- [X] My code follows the code style of this project.
- [X] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/commercetools/theme/middleware/is-authenticated.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
-export default async ({ app, redirect }) => {
-  if (!app.$cookies.get('vsf-commercetools-token')?.scope?.includes('customer_id')) {
+import { Logger } from '@vue-storefront/core';
+
+export default async ({ $vsf, route, redirect }) => {
+  try {
+    const isGuest = await $vsf.$ct.api.isGuest();
```
Comment (filipsobol) [inline packages/commercetools/theme/middleware/is-authenticated.js:5]: Anonymous session is not treated as guest

File: packages/commercetools/api-client/src/api/customerSignOut/index.ts [hunk 2]
```diff
@@ -1,11 +1,7 @@
-const customerSignOut = async ({ config, client }) => {
+const customerSignOut = async ({ config }) => {
```
Comment (filipsobol) [inline packages/commercetools/api-client/src/api/customerSignOut/index.ts:1]: 🤔

File: packages/core/nuxt-theme-module/theme/components/AppHeader.vue [hunk 3]
```diff
@@ -143,7 +143,11 @@ export default {
 
     const accountIcon = computed(() => isAuthenticated.value ? 'profile_fill' : 'profile');
 
-    loadUser();
+    onMounted(() => {
+      // Load only in the browser
+      loadUser();
+      loadCart();
+    });
```
Comment (filipsobol) [inline packages/core/nuxt-theme-module/theme/components/AppHeader.vue:150]: 🤔

File: packages/core/nuxt-theme-module/theme/layouts/default.vue [hunk 4]
```diff
@@ -48,6 +48,25 @@ export default {
...
+    const { load: loadUser } = useUser();
+    const { load: loadCart } = useCart();
+    const { load: loadWishlist } = useWishlist();
+
+    onSSR(async () => {
+      await loadStores();
+    });
+
+    onMounted(() => {
+      console.log('onMounted called');
```
Comment (filipsobol) [inline packages/core/nuxt-theme-module/theme/layouts/default.vue:64]: 🤔

**Review by bloodf — APPROVED** [review]

File: .github/labeler.yml [hunk 5]
```diff
@@ -14,9 +14,10 @@ commercetools:
 boilerplate:
 - packages/boilerplate/**/*
 
-# Add 'docs' label to any change to packages/core/docs files EXCEPT for the changelog
+# Add 'docs' label to any change to packages/core/docs files
 docs:
-- any: ['packages/core/docs/**/*', '!packages/core/docs/changelog/**/*']
+- packages/core/docs/**/*
+- packages/core/docs/.vuepress/**/*
```
Comment (bloodf) [inline .github/labeler.yml:20]: These changes are correct?

**Review by bloodf — APPROVED** [review]

File: .github/labeler.yml (hunk 5 above)
Comment (filipsobol) [reply .github/labeler.yml:20]: I made this change few weeks ago and it popped again when I merged `main`, which means that it got lost somewhere when we moved branches.

**Review by igorwojciechowski — APPROVED** [review]

File: packages/core/nuxt-theme-module/theme/layouts/default.vue [hunk 6]
```diff
@@ -3,16 +3,13 @@
...
+
+    <AppHeader />
 
     <div id="layout">
       <nuxt :key="$route.fullPath"/>
 
-      <LazyHydrate when-visible>
-        <BottomNavigation />
-      </LazyHydrate>
+      <BottomNavigation />
```
Comment (dawid-ziobro) [inline packages/core/nuxt-theme-module/theme/layouts/default.vue:12]: There were some issues with LazyHydrate?

Comment (filipsobol) [reply packages/core/nuxt-theme-module/theme/layouts/default.vue:12]: Yes, but issue will likely be resolved in https://github.com/vuestorefront/vue-storefront/issues/6365

File: packages/commercetools/api-client/src/extensions/emailExtension.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      const token = response?.data?.customerCreatePasswordResetToken?.value;
+
+      const emailObject = {
+        to: email,
+        from: 'password-recovery@vue-storefront.io',
+        subject: `Password recovery for ${email}`,
+        html: `<a href='/reset-password?token=${token}'>Reset your password by clicking this link</a>`
+      };
+
+      console.log(JSON.stringify(emailObject));
```
Comment (dawid-ziobro) [inline packages/commercetools/api-client/src/extensions/emailExtension.ts:19]: Should console.log be here?

File: packages/commercetools/api-client/src/api/isLoggedIn/index.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { isUserSession } from '../../helpers/utils';
+
+const isLoggedIn = (context) => {
+  const { config } = context;
+
+  if (config.handleIsLoggedIn) {
```
Comment (dawid-ziobro) [inline packages/commercetools/api-client/src/api/isLoggedIn/index.ts:6]: It's better to check if `handleIsLoggedIn` type is a function

**Review by dawid-ziobro — APPROVED** [review]

File: packages/commercetools/api-client/src/extensions/emailExtension.ts (hunk 7 above)
Comment (filipsobol) [reply packages/commercetools/api-client/src/extensions/emailExtension.ts:19]: I moved this code from another place and was thinking about removing it, but after speaking with @Baroshem who wrote this, we agree to leave it for now. I don't think it should be here and there is a task to replace it with something better on the theme level (task EN-447).

File: packages/commercetools/api-client/src/api/isLoggedIn/index.ts (hunk 8 above)
Comment (filipsobol) [reply packages/commercetools/api-client/src/api/isLoggedIn/index.ts]: Updated

**Pull request merged**